### PR TITLE
Hide argo rollouts behind feature flag

### PR DESCRIPTION
**Pull request by pnovotnak**
This should suppress warnings seen when argo CRDs are not installed in clusters.

I'm creating this PR as a draft so that tests can be invoked.

### CODE REVIEW

**Comment by pnovotnak** [comment]
I think this is ready to go out now, but I haven't done a ton of testing yet.

**Comment by aattuluri** [comment]
The change overall looks good to me.

Here are a couple of things:
i) Add the rollouts flag as true on the demo install - https://github.com/istio-ecosystem/admiral/blob/399015b580faf670ac0966e38d99e0abebdbc130/install/admiral/overlays/demosinglecluster/envconfig_values.yaml#L21
ii) Try running the integration tests locally (works only for OSX with virtualbox by default) - let me know if you want me to run it.
https://github.com/istio-ecosystem/admiral/blob/master/CONTRIBUTING.md#integration-tests

**Comment by pnovotnak** [comment]
Thanks, will attempt integration tests now

**Comment by pnovotnak** [comment]
Hey sorry for the long delay... I tried a couple of times to get the int tests running locally but was not able to. Would you run it?

**Comment by aattuluri** [comment]
@pnovotnak no worries, I will give it a try over the weekend.

**Comment by pnovotnak** [comment]
Did it pass? When I was poking at it was having an issue with a `minikube` secret which I assume was a cluster access token but couldnt scrounge time to pole at it

**Comment by pnovotnak** [comment]
Also how hard do you think it would be to get this running in ci?

**Comment by aattuluri** [comment]
> Did it pass? When I was poking at it was having an issue with a `minikube` secret which I assume was a cluster access token but couldnt scrounge time to get it figured out

I haven't tried yet, will update you tomorrow

**Comment by aattuluri** [comment]
> Also how hard do you think it would be to get this running in ci?

When I researched I found that the Circle CI host doesn't have the resource limits that will allow a minikube to run istio control plane. If we can find some managed or stanalone k8s cluster generation integration with circle ci we should be able to do it pretty quickly.

The target state would be to have integration tests against more than one hosted cloud accounts (EKS & GKE probably)

**Comment by aattuluri** [comment]
@pnovotnak Ran the test and looks like its throwing a nil pointer reference. I think the block that initializes K8SClient in rollouts.go has been wiped out as part of your PR when you merged it from master. 

Below is the stacktrace.
```
E0728 06:20:36.951616       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 102 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x16fe160, 0x27930d0)
	/Users/aattuluri/go/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/util/runtime/runtime.go:74 +0xa3
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/Users/aattuluri/go/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/util/runtime/runtime.go:48 +0x82
panic(0x16fe160, 0x27930d0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral.(*RolloutController).shouldIgnoreBasedOnLabelsForRollout(0xc0002f2690, 0xc00059c580, 0x1900713)
	/Users/aattuluri/Code/admiral_public/go/src/github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral/rollouts.go:112 +0x14b
github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral.(*RolloutController).Added(0xc0002f2690, 0x18c39a0, 0xc00059c580)
	/Users/aattuluri/Code/admiral_public/go/src/github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral/rollouts.go:164 +0x10c
github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral.(*Controller).processItem(0xc000622510, 0xc000835f80, 0x21, 0x18fb2ee, 0x3, 0x18c39a0, 0xc00059c580, 0x0, 0x0, 0x0, ...)
	/Users/aattuluri/Code/admiral_public/go/src/github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral/controller.go:143 +0xb1
github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral.(*Controller).processNextItem(0xc000622510, 0xc0009a0000)
	/Users/aattuluri/Code/admiral_public/go/src/github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral/controller.go:120 +0x17e
github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral.(*Controller).runWorker(0xc000622510)
	/Users/aattuluri/Code/admiral_public/go/src/github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral/controller.go:107 +0x2b
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc000b97f68)
	/Users/aattuluri/go/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/util/wait/wait.go:152 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000b97f68, 0x12a05f200, 0x0, 0x1, 0xc00032c720)
	/Users/aattuluri/go/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(...)
	/Users/aattuluri/go/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/util/wait/wait.go:88
github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral.(*Controller).Run(0xc000622510, 0xc00032c720)
	/Users/aattuluri/Code/admiral_public/go/src/github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral/controller.go:103 +0x311
created by github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral.NewController
	/Users/aattuluri/Code/admiral_public/go/src/github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral/controller.go:81 +0x430
E0728 06:20:36.960996       1 runtime.go:78] Observed a panic: "invalid memory address or nil pointer dereference" (runtime error: invalid memory address or nil pointer dereference)
goroutine 102 [running]:
k8s.io/apimachinery/pkg/util/runtime.logPanic(0x16fe160, 0x27930d0)
	/Users/aattuluri/go/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/util/runtime/runtime.go:74 +0xa3
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/Users/aattuluri/go/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/util/runtime/runtime.go:48 +0x82
panic(0x16fe160, 0x27930d0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/Users/aattuluri/go/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/util/runtime/runtime.go:55 +0x105
panic(0x16fe160, 0x27930d0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral.(*RolloutController).shouldIgnoreBasedOnLabelsForRollout(0xc0002f2690, 0xc00059c580, 0x1900713)
	/Users/aattuluri/Code/admiral_public/go/src/github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral/rollouts.go:112 +0x14b
github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral.(*RolloutController).Added(0xc0002f2690, 0x18c39a0, 0xc00059c580)
	/Users/aattuluri/Code/admiral_public/go/src/github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral/rollouts.go:164 +0x10c
github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral.(*Controller).processItem(0xc000622510, 0xc000835f80, 0x21, 0x18fb2ee, 0x3, 0x18c39a0, 0xc00059c580, 0x0, 0x0, 0x0, ...)
	/Users/aattuluri/Code/admiral_public/go/src/github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral/controller.go:143 +0xb1
github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral.(*Controller).processNextItem(0xc000622510, 0xc0009a0000)
	/Users/aattuluri/Code/admiral_public/go/src/github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral/controller.go:120 +0x17e
github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral.(*Controller).runWorker(0xc000622510)
	/Users/aattuluri/Code/admiral_public/go/src/github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral/controller.go:107 +0x2b
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc000b97f68)
	/Users/aattuluri/go/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/util/wait/wait.go:152 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000b97f68, 0x12a05f200, 0x0, 0x1, 0xc00032c720)
	/Users/aattuluri/go/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(...)
	/Users/aattuluri/go/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/util/wait/wait.go:88
github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral.(*Controller).Run(0xc000622510, 0xc00032c720)
	/Users/aattuluri/Code/admiral_public/go/src/github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral/controller.go:103 +0x311
created by github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral.NewController
	/Users/aattuluri/Code/admiral_public/go/src/github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral/controller.go:81 +0x430
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0xb0 pc=0x13417cb]

goroutine 102 [running]:
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/Users/aattuluri/go/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/util/runtime/runtime.go:55 +0x105
panic(0x16fe160, 0x27930d0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
k8s.io/apimachinery/pkg/util/runtime.HandleCrash(0x0, 0x0, 0x0)
	/Users/aattuluri/go/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/util/runtime/runtime.go:55 +0x105
panic(0x16fe160, 0x27930d0)
	/usr/local/go/src/runtime/panic.go:679 +0x1b2
github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral.(*RolloutController).shouldIgnoreBasedOnLabelsForRollout(0xc0002f2690, 0xc00059c580, 0x1900713)
	/Users/aattuluri/Code/admiral_public/go/src/github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral/rollouts.go:112 +0x14b
github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral.(*RolloutController).Added(0xc0002f2690, 0x18c39a0, 0xc00059c580)
	/Users/aattuluri/Code/admiral_public/go/src/github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral/rollouts.go:164 +0x10c
github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral.(*Controller).processItem(0xc000622510, 0xc000835f80, 0x21, 0x18fb2ee, 0x3, 0x18c39a0, 0xc00059c580, 0x0, 0x0, 0x0, ...)
	/Users/aattuluri/Code/admiral_public/go/src/github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral/controller.go:143 +0xb1
github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral.(*Controller).processNextItem(0xc000622510, 0xc0009a0000)
	/Users/aattuluri/Code/admiral_public/go/src/github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral/controller.go:120 +0x17e
github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral.(*Controller).runWorker(0xc000622510)
	/Users/aattuluri/Code/admiral_public/go/src/github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral/controller.go:107 +0x2b
k8s.io/apimachinery/pkg/util/wait.JitterUntil.func1(0xc000b97f68)
	/Users/aattuluri/go/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/util/wait/wait.go:152 +0x5e
k8s.io/apimachinery/pkg/util/wait.JitterUntil(0xc000b97f68, 0x12a05f200, 0x0, 0x1, 0xc00032c720)
	/Users/aattuluri/go/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/util/wait/wait.go:153 +0xf8
k8s.io/apimachinery/pkg/util/wait.Until(...)
	/Users/aattuluri/go/pkg/mod/k8s.io/apimachinery@v0.17.3/pkg/util/wait/wait.go:88
github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral.(*Controller).Run(0xc000622510, 0xc00032c720)
	/Users/aattuluri/Code/admiral_public/go/src/github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral/controller.go:103 +0x311
created by github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral.NewController
	/Users/aattuluri/Code/admiral_public/go/src/github.com/istio-ecosystem/admiral/admiral/pkg/controller/admiral/controller.go:81 +0x430
```

File: admiral/pkg/controller/admiral/rollouts.go [hunk 1]
```diff
@@ -140,13 +140,6 @@ func NewRolloutsController(stopCh <-chan struct{}, handler RolloutHandler, confi
 		return nil, fmt.Errorf("failed to create rollouts controller argo client: %v", err)
 	}
 
-	roController.K8sClient, err = K8sClientFromConfig(config)
-	if err != nil {
-		return nil, fmt.Errorf("failed to create rollouts controller k8s client: %v", err)
-	}
-
-	roController.RolloutClient = rolloutClient.ArgoprojV1alpha1()
-
```
Comment (aattuluri) [inline admiral/pkg/controller/admiral/rollouts.go:149]: This block should be restored, it's causing a nil pointer reference.

**Comment by aattuluri** [comment]
Also please add the following below [this](https://github.com/istio-ecosystem/admiral/blob/ebebec2a49dd742fd12d60fc0f55b03cb3192ccf/install/admiral/overlays/demosinglecluster/envconfig_values.yaml#L21) line:
`- --argo_rollouts=true`

**Comment by aattuluri** [comment]
@pnovotnak Did you get a chance to look at my comments?

**Comment by pnovotnak** [comment]
Sorry for the force push. I moved machines and forgot my `prepare-commit-msg` hook

**Pull request merged**